src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp

```diff
@@ -1571,31 +1577,31 @@ GadgetronImageData::multiply(const DataContainer& a_x, const DataContainer& a_y)
 {
 	SIRF_DYNAMIC_CAST(const GadgetronImageData, x, a_x);
 	SIRF_DYNAMIC_CAST(const GadgetronImageData, y, a_y);
-    binary_op(x, y, DataContainer::product<complex_float_t>);
+    binary_op(x, y, DataContainerTempl<complex_float_t>::product);
```

Comment:
I think all these would be a lot cleaner if we'd have
```
typedef GadgetronImageData self_type;
```
as then we can do
```
    SIRF_DYNAMIC_CAST(const  self_type, x, a_x);
    SIRF_DYNAMIC_CAST(const self_type, y, a_y);
    binary_op(x, y, self_type::product);
```
such that the code looks class-independent

Reply:
@KrisThielemans works nicely, thanks! Will apply to all container methods after I got rid of all `void*` (operations involving scalars still have them).